internal/blockchain/ethereum/ethereum.go

```diff
@@ -284,19 +288,20 @@ func (e *Ethereum) handleContractEvent(ctx context.Context, msgJSON fftypes.JSON
 	}
 	delete(msgJSON, "data")
 
-	event := &blockchain.ContractEvent{
-		Subscription: sub,
+	event := &blockchain.EventWithContext{
+		Subscription:   sub,
+		BlockchainTXID: sTransactionHash,
```

Comment:
Is this used in the event handler for contract events?

Reply:
Ok, good for now. Will need to have a think about whether we should be putting it somewhere on that path.